apf/apf.h

```diff
@@ -718,15 +714,32 @@ void unfreeze(Field* f);
 /** \brief Returns true iff the Field uses array storage. */
 bool isFrozen(Field* f);
 
-template <typename T>
-T* getArrayDataT(Field * f);
+/** \brief Return the contiguous array storing this field.
+  \details This function is only defined for fields
+  which are using array storage, for which apf::isFrozen
+  returns true.
+  \note If the underlying field data type is NOT double,
+  this will cause an assert fail in all compile modes.
+ */
+double* getArrayData(Field* f);
```

Comment:
Do we have a mechanism for depreciation? If so, this should probably be depreciated and replaced with `getDoubleArrayData` to have a consistent API.

Reply:
Since C++14 we have [[deprecated]], though this codebase is explicitly only up to C++11 IIRC and we haven't put in a deprecation mechanism as far as I know.

C++14 was basically just a small step from 11 so moving core to that should be possible, but is a different issue. 

For the time being I will introduce a more consistent API function and not in the doxygen this is deprecated (see apfNumbering.h near the bottom for the only other place this has been done I think).